ansi/halfblocks/encoder.go

```diff
@@ -0,0 +1,626 @@
+package halfblocks
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+	"math"
+	"strings"
+)
+
+// Block represents different Unicode block characters
+type Block struct {
+	Char        rune
+	Coverage    []bool // Which parts of the block are filled (true = filled)
```

Comment:
This is always gonna be 4 sides
```suggestion
	Coverage    [4]bool // Which parts of the block are filled (true = filled)
```